google-account-plugin/src/com/google/cloud/tools/intellij/stats/UsageTracker.java

```diff
@@ -33,4 +34,26 @@ void trackEvent(@NotNull String eventCategory,
       @Nullable String eventLabel,
       @Nullable Integer eventValue);
 
+  PartialTrackingEventAction trackEvent(String category);
+
+  interface SendsEvent {
+
+    void send();
+  }
+
+
+  interface PartialTrackingEventAction extends SendsEvent {
+
+    PartialTrackingEventLabel withAction(String value);
+
+    interface PartialTrackingEventLabel extends SendsEvent {
+
+      PartialTrackingEventValue andLabel(String value);
+
+      interface PartialTrackingEventValue extends SendsEvent {
```

Comment:
Yeah. each nested interface will never be constructed outside of the
context of its parent.  i.e. the parent must be specified before you can
specify the child.

On Wed, Jun 22, 2016 at 1:43 PM, Etan Shaul <notifications@github.com>
wrote:

> In
> google-account-plugin/src/com/google/cloud/tools/intellij/stats/UsageTracker.java
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/752#discussion_r68099959>
> :
>
> > +
> > +  interface SendsEvent {
> > +
> > +    void send();
> > +  }
> > +
> > +
> > +  interface PartialTrackingEventAction extends SendsEvent {
> > +
> > +    PartialTrackingEventLabel withAction(String value);
> > +
> > +    interface PartialTrackingEventLabel extends SendsEvent {
> > +
> > +      PartialTrackingEventValue andLabel(String value);
> > +
> > +      interface PartialTrackingEventValue extends SendsEvent {
>
> any reason why this is nested?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/752/files/baccf888600f472b69946034e2bc213cc9ce67ec#r68099959>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHf5HXOQz9ikww61Yrl-FlnZkqUflaUfks5qOXQ5gaJpZM4I8B5a>
> .
>